File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam6And7April2019/04.CinemaVoucher.js
function cinemaVoucher(input) {
    let index = 0;
    let voucherValue = Number(input[index]);
    index++;
    command = input[index];
    index++;

    let ticketsCount = 0;
    let productCount = 0;

    while(command !== 'End') {
        let productType = command;
        let currentProductPrice = 0;
        if(productType.length > 8) {
            let firstSymbol = productType.charCodeAt([0]);
            let secondSymbol = productType.charCodeAt([1]);
            currentProductPrice = firstSymbol + secondSymbol;

            if (currentProductPrice <= voucherValue) {
                ticketsCount++;
                voucherValue -= currentProductPrice;
            } else {
                break;
            }
        } else {
            currentProductPrice = productType.charCodeAt([0]);
            if (currentProductPrice <= voucherValue) {
                productCount++;
                voucherValue -= currentProductPrice;
            } else {
                break;
            }
        }

        command = input[index];
        index++;
    }

    console.log(ticketsCount);
    console.log(productCount);
}

cinemaVoucher(["300",
"Captain Marvel",
"popcorn",
"Pepsi"])

cinemaVoucher(["1500",
"Avengers: Endgame",
"Bohemian Rhapsody",
"Deadpool 2",
"End"])

File: Programming_Basics_with_Python/programming_basics_online_exam_6_and_7_april_2019/04._cinema_voucher.py
voucher_amount = int(input())
command = input()

tickets_count = 0
other_products_count = 0
while not command == 'End':
    product = command
    current_product_price = 0
    is_ticket = False

    if len(product) > 8:
        is_ticket = True
        current_product_price = ord(product[0]) + ord(product[1])
    else:
        current_product_price = ord(product[0])

    if voucher_amount - current_product_price >= 0:
        voucher_amount -= current_product_price
        if is_ticket:
            tickets_count += 1
        else:
            other_products_count += 1
    else:
        break

    command = input()

print(tickets_count)
print(other_products_count)
